app/components/pay-page/choose-membership-plan-modal.ts

```diff
@@ -0,0 +1,134 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import fade from 'ember-animated/transitions/fade';
+import { inject as service } from '@ember/service';
+import Store from '@ember-data/store';
+import window from 'ember-window-mock';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type AuthenticatorService from 'codecrafters-frontend/services/authenticator';
+
+interface Signature {
+  Element: HTMLAnchorElement;
+
+  Args: {
+    onClose: () => void;
+    regionalDiscount?: RegionalDiscountModel | null;
+    shouldApplyRegionalDiscount?: boolean;
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
+  };
+}
+
+export default class ChooseMembershipPlanModal extends Component<Signature> {
+  @service declare authenticator: AuthenticatorService;
+  @service declare store: Store;
+
+  transition = fade;
+
+  @tracked extraInvoiceDetailsRequested = false;
+  @tracked isCreatingCheckoutSession = false;
+  @tracked previewType: 'plan' | 'invoice-details' = 'plan';
```

Comment:
```suggestion
  @tracked previewType: 'plan' | 'invoice-details' = 'plan';
```

@Arpan-206 I don't put a lot of thought into variable names etc. in design previews, upto you to clean them up. `previewType` doesn't really make sense here - check other wizard-like flows and I think we use `currentStep` in most of them. The `teams/pay` one is one example of a wizard flow, the onboarding survey is another.